sdk/python/kfp_tekton/compiler/_op_to_template.py

```diff
@@ -505,4 +532,59 @@ def _op_to_template(op: BaseOp,
             template.setdefault('metadata', {}).setdefault('annotations', {})['pipelines.kubeflow.org/max_cache_staleness'] = \
                 str(op.execution_options.caching_strategy.max_cache_staleness)
 
+    # Sort and arrange results based on provided estimate size and process results in multi-steps if the result sizes are too big.
+    result_size_map = "{}"
+    if processed_op.pod_annotations:
+        result_size_map = processed_op.pod_annotations.get("tekton-result-sizes", "{}")
+    # Only sort and arrange results when the estimated sizes are given.
+    if result_size_map and result_size_map != "{}":
+        try:
+            result_size_map = json.loads(result_size_map)
+        except ValueError:
+            raise("tekton-result-sizes annotation is not a valid JSON")
+        # Sort key orders based on values
+        result_size_map = dict(sorted(result_size_map.items(), key=lambda item: item[1]))
+        max_byte_size = 2048
+        verified_result_size_map = {0: {}}
+        op_result_names = [name['name'] for name in template['spec']['results']]
+        step_bins = {0: 0}
+        step_counter = 0
+        # Group result files to not exceed max_byte_size
+        # TODO: group result files with better bin packing algorithm
+        for key, value in result_size_map.items():
+            try:
+                value = int(value)
+            except ValueError:
+                raise("Estimated value for result %s is %s, but it needs to be an integer." % (key, value))
+            if key in op_result_names:
```

Comment:
No warning or anything if there is no such result --- it's a bit misleading. Between KFP, Argo, Tekton etc, `-` and `_` are often used interchangeably, so it could be surprising if it simply doesn't work when one is used instead of the other.

Reply:
done, adding new warnings and test cases